File: src/java/eu/sig/training/ch03/FlagFactory.java
package eu.sig.training.ch03;

import java.awt.Color;
import java.util.Arrays;
import java.util.List;

public class FlagFactory {

    // tag::getFlag[]
    public List<Color> getFlagColors(Nationality nationality) {
        List<Color> result;
        switch (nationality) {
        case DUTCH:
            result = Arrays.asList(Color.RED, Color.WHITE, Color.BLUE);
            break;
        case GERMAN:
            result = Arrays.asList(Color.BLACK, Color.RED, Color.YELLOW);
            break;
        case BELGIAN:
            result = Arrays.asList(Color.BLACK, Color.YELLOW, Color.RED);
            break;
        case FRENCH:
            result = Arrays.asList(Color.BLUE, Color.WHITE, Color.RED);
            break;
        case ITALIAN:
            result = Arrays.asList(Color.GREEN, Color.WHITE, Color.RED);
            break;
        case UNCLASSIFIED:
        default:
            result = Arrays.asList(Color.GRAY);
            break;
        }
        return result;
    }
    // end::getFlag[]

}

File: src/csharp/eu/sig/training/ch03/FlagFactory.cs
using System.Collections.Generic;
using System.Drawing;

namespace eu.sig.training.ch03
{

    public class FlagFactory
    {
        // tag::getFlag[]
        public IList<Color> GetFlagColors(Nationality nationality)
        {
            List<Color> result;
            switch (nationality)
            {
                case Nationality.DUTCH:
                    result = new List<Color> { Color.Red, Color.White, Color.Blue };
                    break;
                case Nationality.GERMAN:
                    result = new List<Color> { Color.Black, Color.Red, Color.Yellow };
                    break;
                case Nationality.BELGIAN:
                    result = new List<Color> { Color.Black, Color.Yellow, Color.Red };
                    break;
                case Nationality.FRENCH:
                    result = new List<Color> { Color.Blue, Color.White, Color.Red };
                    break;
                case Nationality.ITALIAN:
                    result = new List<Color> { Color.Green, Color.White, Color.Red };
                    break;
                case Nationality.UNCLASSIFIED:
                default:
                    result = new List<Color> { Color.Gray };
                    break;
            }
            return result;
        }
        // end::getFlag[]
    }

}
